src/test/scala/com/databricks/spark/csv/CsvWriteSuite.scala

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2014 Databricks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.databricks.spark.csv
+
+import java.io.File
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.{FileSystem, FileUtil, Path}
+import org.apache.spark.SparkContext
+import org.apache.spark.sql.types._
+import org.apache.spark.sql.{DataFrame, SQLContext}
+import org.scalatest.{BeforeAndAfterAll, FunSuite}
+
+import scala.io.Source
+
+class CsvWriteSuite extends FunSuite with BeforeAndAfterAll {
+
+  val datesFile = "src/test/resources/dates.csv"
+  val tempEmptyDir = "target/test/empty/"
+
+  def parserLib: String = "COMMONS"
```

Comment:
Are the parser libs used when writing? These tests are used to test writing, not reading of the file. It does not matter which reader is used (as long as it works). Testing the reader with different parser libs is the job of CsvSuite not CsvWriteSuite. So is this subclassing necessary?